### ISSUE TITLE
JAMSVG crashing for color gradients without "style" info.

### ISSUE THREAD

**Issue by mark-whcc** [jmenter/JAMSVGImage#14]
I have a SVG Image that was generated by Photoshop that crashes JAMSVGImage:
https://gist.github.com/mark-whcc/1e28bd3420d0d21ac63d

The crash comes at line 78 of `JAMStylizedBezierPath.m` when it tries to add a color to an NSArray, but the color is nil.

It looks like there is a problem in `JAMStylizedBezierPathFactory.m` that would cause this, but this code is never hit. `parseStyleColor:` is scanning for `stop-color:` but my sample file has `stop-color=`. It will have the same problem with `stop-opacity:`.

The actual problem is `parseStyleColor:` is being called with no `"style"`  (Line 89 of `JAMStylizedBezierPathFactory.m `) key in the dictionary, so the color, which is in the file, is not being found. This leaves the color property of `JAMSVGGradientColorStop` nil.

**Comment by jmenter** [comment]
Yeah, I see what's going on. I fixed the stop-color problems but the stop positions are expressed in percentages of the bounding box of the element the gradient is being applied to. That will take some work.

I didn't even know you could export SVG out of Photoshop. I have CS6 and don't see the option.

**Comment by mark-whcc** [comment]
The Photoshop export isn't a direct option, it's part of the generator plugin introduced in Photoshop CC. We clean up the JSON output from the generator, but don't change the SVG files. The card editing interface in our app is entirely generated from Photoshop generator plugin output.

http://blogs.adobe.com/photoshopdotcom/2013/09/introducing-adobe-generator-for-photoshop-cc.html

The designers aren't supposed to use gradients, but this one slipped through. They are even open to discontinuing that card since it doesn't sound like you will support this style of gradient.

Thanks for your help.
